### PR TITLE
rgw: OpsLogFile::stop() signals under mutex

### PR DESCRIPTION
**Pull request by cbodley**
this shuts up `ceph::debug_condition_variable`'s assertion that the associated mutex is held during `notify_one()`. this is not strictly required for correct use, but is a common source of bugs

Fixes: https://tracker.ceph.com/issues/55432

<details>
<summary>Show available Jenkins commands</summary>

- `jenkins retest this please`
- `jenkins test classic perf`
- `jenkins test crimson perf`
- `jenkins test signed`
- `jenkins test make check`
- `jenkins test make check arm64`
- `jenkins test submodules`
- `jenkins test dashboard`
- `jenkins test dashboard cephadm`
- `jenkins test api`
- `jenkins test docs`
- `jenkins render docs`
- `jenkins test ceph-volume all`
- `jenkins test ceph-volume tox`
- `jenkins test windows`
</details>

### CODE REVIEW

**Comment by cbodley** [comment]
the second commit isn't necessary for the bug fix, but i think the simplification makes OpsLogFile's locking much easier to reason about

File: src/rgw/rgw_log.cc [hunk 1]
```diff
@@ -410,8 +409,8 @@ void OpsLogFile::start() {
 
 void OpsLogFile::stop() {
   {
-    std::unique_lock lock(log_mutex);
-    cond_flush.notify_one();
+    std::unique_lock lock(mutex);
+    cond.notify_one();
```
Comment (mattbenjamin) [inline src/rgw/rgw_log.cc:413]: should stopped be set before the cond.notify_one()?

**Review by mattbenjamin — APPROVED** [review]
+1

**Comment by cbodley** [comment]
```
```

File: src/rgw/rgw_log.cc (hunk 1 above)
Comment (cfsnyder) [reply src/rgw/rgw_log.cc:414]: Looks like it should be fine since the mutex is still held while stopped is set

File: src/rgw/rgw_log.cc [hunk 2]
```diff
@@ -389,16 +388,17 @@ void OpsLogFile::flush()
...
     if (!log_buffer.empty()) {
       lock.unlock();
       flush();
       lock.lock();
       continue;
     }
-    cond_flush.wait(lock);
+    cond.wait(lock);
   }
+  lock.unlock();
```
Comment (cfsnyder) [inline src/rgw/rgw_log.cc:401]: :+1:

**Review by cfsnyder — APPROVED** [review]

**Comment by cbodley** [comment]
jenkins test make check

**Comment by cbodley** [comment]
passed qa in https://pulpito.ceph.com/cbodley-2022-04-26_01:02:21-rgw-wip-cbodley-testing-distro-default-smithi/

**Pull request merged**